src/Components/Endpoints/src/Builder/RazorComponentEndpointConventionBuilderExtensions.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Builder;
+
+/// <summary>
+/// Configures which assemblies are part of the given Razor Component Application.
+/// </summary>
+public static class RazorComponentEndpointConventionBuilderExtensions
+{
+    /// <summary>
+    /// Adds the given assemblies to the component application.
+    /// </summary>
+    /// <param name="builder">The <see cref="RazorComponentEndpointConventionBuilder"/>.</param>"/>
+    /// <param name="assemblies">The <see cref="Assembly"/> instances to add.</param>
+    /// <returns>The <see cref="RazorComponentEndpointConventionBuilder"/>.</returns>
+    /// <remarks>
+    /// The provided assemblies will be scanned for pages that will be mapped as endpoints.
+    /// </remarks>
+    public static RazorComponentEndpointConventionBuilder AddAssemblies(
```

Comment:
Did we settle on `AddAssemblies`? That's my preference and what I thought we agreed on, but you were sharing your screen, and your most recent edit of the proposal #49756 showed `AddAdditionalAssemblies`. I can update https://github.com/dotnet/aspnetcore/issues/49756#issuecomment-1660930751. I just want to make sure this is intentional.

Reply:
I don't understand that implication at all. We don't call `AddSingleton` `AddAdditionalSingleton` just because there are some default singleton services like `IServiceProvider`.

That said, I said basically this during API review, and I think the consensus was that we should add `Additional` to the method name to make it super clear that the main assembly is included by default.